pyworkflow/em/protocol/protocol_movies.py

```
@@ -102,14 +102,73 @@ def _insertAllSteps(self):
         # inserting each of the steps for each movie
         self.insertedDict = {}
         self.samplingRate = self.inputMovies.get().getSamplingRate()
-        
+
+        # Gain and Dark conversion step
+        self.convertCIStep = []
+        convertStepId = self._insertFunctionStep('_convertCorrectionImages',
+                                                  prerequisites=[])
+        self.convertCIStep.append(convertStepId)
+
         # Conversion step is part of processMovieStep because of streaming.
         movieSteps = self._insertNewMoviesSteps(self.insertedDict,
                                                 self.inputMovies.get())
         finalSteps = self._insertFinalSteps(movieSteps)
         self._insertFunctionStep('createOutputStep',
                                  prerequisites=finalSteps, wait=True)
 
+    # STEP to convert correction images if apply
+    def _convertCorrectionImages(self):
+
+        movs = self.inputMovies.get()
+
+        # Convert gain
+        gain = movs.getGain()
+        movs.setGain(self.__convertCorrectionImage(gain))
+
+        # Convert dark
+        dark = movs.getDark()
+        movs.setDark(self.__convertCorrectionImage(dark))
+
+    def __convertCorrectionImage(self, correctionImage):
+        """ Will convert a gain or dark file to a compatible one and return
+        the final file name. Otherwise, will return same passed parameter"""
+
+        if correctionImage is None:
+            return
+
+        elif not os.path.exists(correctionImage):
```

It the gain is set and does not exist...the protocol can not just simply ignore it, I suggest we need to raise an Exception in order to notify the user that something is wrong: either remove the gain parameter or copy the file to the expected path.